### PR TITLE
Add state files for junos module and unit test dependencies

### PR DESCRIPTION
**Pull request by rajvidhimar**
This pull request adds state files to install jxmlease and junos-eznc on jenkins which are required by the junos module and the junos unit-tests.

### CODE REVIEW

**Comment by rajvidhimar** [comment]
This is with reference to the following PR on salt:
https://github.com/saltstack/salt/pull/39884

**Comment by rallytime** [comment]
Hi @rajvidhimar - I ran this through a quick smoke test to see if the states will run correctly on our test VMS and it looks like one of the states here is failing. Here's the output from the state run:
```
----------
          ID: junos-eznc
    Function: pip.installed
      Result: False
     Comment: Failed to install packages: junos-eznc. Error: Collecting junos-eznc
                Downloading junos-eznc-2.0.1.zip (115kB)
              Collecting lxml>=3.2.4 (from junos-eznc)
                Downloading lxml-3.7.3-cp27-cp27mu-manylinux1_x86_64.whl (6.8MB)
              Collecting ncclient>=0.4.6 (from junos-eznc)
                Downloading ncclient-0.5.3.tar.gz (63kB)
              Collecting paramiko>=1.15.2 (from junos-eznc)
                Downloading paramiko-2.1.2-py2.py3-none-any.whl (172kB)
              Collecting scp>=0.7.0 (from junos-eznc)
                Downloading scp-0.10.2-py2.py3-none-any.whl
              Requirement already satisfied: jinja2>=2.7.1 in /usr/lib/python2.7/site-packages (from junos-eznc)
              Requirement already satisfied: PyYAML>=3.10 in /usr/lib64/python2.7/site-packages (from junos-eznc)
              Collecting netaddr (from junos-eznc)
                Downloading netaddr-0.7.19-py2.py3-none-any.whl (1.6MB)
              Requirement already satisfied: six in /usr/lib/python2.7/site-packages (from junos-eznc)
              Collecting pyserial (from junos-eznc)
                Downloading pyserial-3.3-py2.py3-none-any.whl (189kB)
              Requirement already satisfied: setuptools>0.6 in /usr/lib/python2.7/site-packages (from ncclient>=0.4.6->junos-eznc)
              Collecting cryptography>=1.1 (from paramiko>=1.15.2->junos-eznc)
                Downloading cryptography-1.8.1.tar.gz (423kB)
              Collecting pyasn1>=0.1.7 (from paramiko>=1.15.2->junos-eznc)
                Downloading pyasn1-0.2.3-py2.py3-none-any.whl (53kB)
              Requirement already satisfied: MarkupSafe>=0.23 in /usr/lib64/python2.7/site-packages (from jinja2>=2.7.1->junos-eznc)
              Requirement already satisfied: appdirs>=1.4.0 in /usr/lib/python2.7/site-packages (from setuptools>0.6->ncclient>=0.4.6->junos-eznc)
              Requirement already satisfied: packaging>=16.8 in /usr/lib/python2.7/site-packages (from setuptools>0.6->ncclient>=0.4.6->junos-eznc)
              Collecting idna>=2.1 (from cryptography>=1.1->paramiko>=1.15.2->junos-eznc)
                Downloading idna-2.5-py2.py3-none-any.whl (55kB)
              Collecting asn1crypto>=0.21.0 (from cryptography>=1.1->paramiko>=1.15.2->junos-eznc)
                Downloading asn1crypto-0.22.0-py2.py3-none-any.whl (97kB)
              Collecting enum34 (from cryptography>=1.1->paramiko>=1.15.2->junos-eznc)
                Downloading enum34-1.1.6-py2-none-any.whl
              Requirement already satisfied: ipaddress in /usr/lib/python2.7/site-packages (from cryptography>=1.1->paramiko>=1.15.2->junos-eznc)
              Collecting cffi>=1.4.1 (from cryptography>=1.1->paramiko>=1.15.2->junos-eznc)
                Downloading cffi-1.10.0-cp27-cp27mu-manylinux1_x86_64.whl (392kB)
              Requirement already satisfied: pyparsing in /usr/lib/python2.7/site-packages (from packaging>=16.8->setuptools>0.6->ncclient>=0.4.6->junos-eznc)
              Collecting pycparser (from cffi>=1.4.1->cryptography>=1.1->paramiko>=1.15.2->junos-eznc)
                Downloading pycparser-2.17.tar.gz (231kB)
              Building wheels for collected packages: junos-eznc, ncclient, cryptography, pycparser
                Running setup.py bdist_wheel for junos-eznc: started
                Running setup.py bdist_wheel for junos-eznc: finished with status 'done'
                Stored in directory: /root/.cache/pip/wheels/f6/01/a2/b2f502d90b24cbf2512d86710deb76f0d0e02a46502f4a0110
                Running setup.py bdist_wheel for ncclient: started
                Running setup.py bdist_wheel for ncclient: finished with status 'done'
                Stored in directory: /root/.cache/pip/wheels/86/30/68/153d65b60834981c1960737f3f2de488574ba5355fe1329558
                Running setup.py bdist_wheel for cryptography: started
                Running setup.py bdist_wheel for cryptography: finished with status 'error'
                Complete output from command /usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-ux0MHT/cryptography/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmp6xO1Kcpip-wheel- --python-tag cp27:
                Package libffi was not found in the pkg-config search path.
                Perhaps you should add the directory containing `libffi.pc'
                to the PKG_CONFIG_PATH environment variable
                No package 'libffi' found
                Package libffi was not found in the pkg-config search path.
                Perhaps you should add the directory containing `libffi.pc'
                to the PKG_CONFIG_PATH environment variable
                No package 'libffi' found
                Package libffi was not found in the pkg-config search path.
                Perhaps you should add the directory containing `libffi.pc'
                to the PKG_CONFIG_PATH environment variable
                No package 'libffi' found
                Package libffi was not found in the pkg-config search path.
                Perhaps you should add the directory containing `libffi.pc'
                to the PKG_CONFIG_PATH environment variable
                No package 'libffi' found
                Package libffi was not found in the pkg-config search path.
                Perhaps you should add the directory containing `libffi.pc'
                to the PKG_CONFIG_PATH environment variable
                No package 'libffi' found
                c/_cffi_backend.c:15:17: fatal error: ffi.h: No such file or directory
                 #include <ffi.h>
                                 ^
                compilation terminated.
                Traceback (most recent call last):
                  File "<string>", line 1, in <module>
                  File "/tmp/pip-build-ux0MHT/cryptography/setup.py", line 335, in <module>
                    **keywords_with_side_effects(sys.argv)
                  File "/usr/lib64/python2.7/distutils/core.py", line 112, in setup
                    _setup_distribution = dist = klass(attrs)
                  File "/usr/lib/python2.7/site-packages/setuptools/dist.py", line 317, in __init__
                    self.fetch_build_eggs(attrs['setup_requires'])
                  File "/usr/lib/python2.7/site-packages/setuptools/dist.py", line 372, in fetch_build_eggs
                    replace_conflicting=True,
                  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 851, in resolve
                    dist = best[req.key] = env.best_match(req, ws, installer)
                  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1123, in best_match
                    return self.obtain(req, installer)
                  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1135, in obtain
                    return installer(requirement)
                  File "/usr/lib/python2.7/site-packages/setuptools/dist.py", line 440, in fetch_build_egg
                    return cmd.easy_install(req)
                  File "/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 674, in easy_install
                    return self.install_item(spec, dist.location, tmpdir, deps)
                  File "/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 700, in install_item
                    dists = self.install_eggs(spec, download, tmpdir)
                  File "/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 881, in install_eggs
                    return self.build_and_install(setup_script, setup_base)
                  File "/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 1120, in build_and_install
                    self.run_setup(setup_script, setup_base, args)
                  File "/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 1108, in run_setup
                    raise DistutilsError("Setup script exited with %s" % (v.args[0],))
                distutils.errors.DistutilsError: Setup script exited with error: command 'gcc' failed with exit status 1

                ----------------------------------------
                Running setup.py clean for cryptography
                Complete output from command /usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-ux0MHT/cryptography/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" clean --all:
                Package libffi was not found in the pkg-config search path.
                Perhaps you should add the directory containing `libffi.pc'
                to the PKG_CONFIG_PATH environment variable
                No package 'libffi' found
                Package libffi was not found in the pkg-config search path.
                Perhaps you should add the directory containing `libffi.pc'
                to the PKG_CONFIG_PATH environment variable
                No package 'libffi' found
                Package libffi was not found in the pkg-config search path.
                Perhaps you should add the directory containing `libffi.pc'
                to the PKG_CONFIG_PATH environment variable
                No package 'libffi' found
                Package libffi was not found in the pkg-config search path.
                Perhaps you should add the directory containing `libffi.pc'
                to the PKG_CONFIG_PATH environment variable
                No package 'libffi' found
                Package libffi was not found in the pkg-config search path.
                Perhaps you should add the directory containing `libffi.pc'
                to the PKG_CONFIG_PATH environment variable
                No package 'libffi' found
                c/_cffi_backend.c:15:17: fatal error: ffi.h: No such file or directory
                 #include <ffi.h>
                                 ^
                compilation terminated.
                Traceback (most recent call last):
                  File "<string>", line 1, in <module>
                  File "/tmp/pip-build-ux0MHT/cryptography/setup.py", line 335, in <module>
                    **keywords_with_side_effects(sys.argv)
                  File "/usr/lib64/python2.7/distutils/core.py", line 112, in setup
                    _setup_distribution = dist = klass(attrs)
                  File "/usr/lib/python2.7/site-packages/setuptools/dist.py", line 317, in __init__
                    self.fetch_build_eggs(attrs['setup_requires'])
                  File "/usr/lib/python2.7/site-packages/setuptools/dist.py", line 372, in fetch_build_eggs
                    replace_conflicting=True,
                  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 851, in resolve
                    dist = best[req.key] = env.best_match(req, ws, installer)
                  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1123, in best_match
                    return self.obtain(req, installer)
                  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1135, in obtain
                    return installer(requirement)
                  File "/usr/lib/python2.7/site-packages/setuptools/dist.py", line 440, in fetch_build_egg
                    return cmd.easy_install(req)
                  File "/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 674, in easy_install
                    return self.install_item(spec, dist.location, tmpdir, deps)
                  File "/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 700, in install_item
                    dists = self.install_eggs(spec, download, tmpdir)
                  File "/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 881, in install_eggs
                    return self.build_and_install(setup_script, setup_base)
                  File "/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 1120, in build_and_install
                    self.run_setup(setup_script, setup_base, args)
                  File "/usr/lib/python2.7/site-packages/setuptools/command/easy_install.py", line 1108, in run_setup
                    raise DistutilsError("Setup script exited with %s" % (v.args[0],))
                distutils.errors.DistutilsError: Setup script exited with error: command 'gcc' failed with exit status 1

                ----------------------------------------
                Running setup.py bdist_wheel for pycparser: started
                Running setup.py bdist_wheel for pycparser: finished with status 'done'
                Stored in directory: /root/.cache/pip/wheels/a8/0b/41/dc95621f9d3a0da7bc191b8a71f0e8182ffd3cc5f33ac55005
              Successfully built junos-eznc ncclient pycparser
              Failed to build cryptography
              Installing collected packages: lxml, idna, asn1crypto, enum34, pycparser, cffi, cryptography, pyasn1, paramiko, ncclient, scp, netaddr, pyserial, junos-eznc
                Running setup.py install for cryptography: started
                  Running setup.py install for cryptography: finished with status 'error'
                  Complete output from command /usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-ux0MHT/cryptography/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-MmCJjn-record/install-record.txt --single-version-externally-managed --compile:
                  running install
                  running build
                  running build_py
                  creating build
                  creating build/lib.linux-x86_64-2.7
                  creating build/lib.linux-x86_64-2.7/cryptography
                  copying src/cryptography/__init__.py -> build/lib.linux-x86_64-2.7/cryptography
                  copying src/cryptography/fernet.py -> build/lib.linux-x86_64-2.7/cryptography
                  copying src/cryptography/exceptions.py -> build/lib.linux-x86_64-2.7/cryptography
                  copying src/cryptography/utils.py -> build/lib.linux-x86_64-2.7/cryptography
                  copying src/cryptography/__about__.py -> build/lib.linux-x86_64-2.7/cryptography
                  creating build/lib.linux-x86_64-2.7/cryptography/hazmat
                  copying src/cryptography/hazmat/__init__.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat
                  creating build/lib.linux-x86_64-2.7/cryptography/x509
                  copying src/cryptography/x509/oid.py -> build/lib.linux-x86_64-2.7/cryptography/x509
                  copying src/cryptography/x509/__init__.py -> build/lib.linux-x86_64-2.7/cryptography/x509
                  copying src/cryptography/x509/extensions.py -> build/lib.linux-x86_64-2.7/cryptography/x509
                  copying src/cryptography/x509/name.py -> build/lib.linux-x86_64-2.7/cryptography/x509
                  copying src/cryptography/x509/base.py -> build/lib.linux-x86_64-2.7/cryptography/x509
                  copying src/cryptography/x509/general_name.py -> build/lib.linux-x86_64-2.7/cryptography/x509
                  creating build/lib.linux-x86_64-2.7/cryptography/hazmat/bindings
                  copying src/cryptography/hazmat/bindings/__init__.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/bindings
                  creating build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives
                  copying src/cryptography/hazmat/primitives/serialization.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives
                  copying src/cryptography/hazmat/primitives/hashes.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives
                  copying src/cryptography/hazmat/primitives/constant_time.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives
                  copying src/cryptography/hazmat/primitives/__init__.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives
                  copying src/cryptography/hazmat/primitives/keywrap.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives
                  copying src/cryptography/hazmat/primitives/hmac.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives
                  copying src/cryptography/hazmat/primitives/padding.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives
                  copying src/cryptography/hazmat/primitives/cmac.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives
                  creating build/lib.linux-x86_64-2.7/cryptography/hazmat/backends
                  copying src/cryptography/hazmat/backends/interfaces.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends
                  copying src/cryptography/hazmat/backends/__init__.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends
                  copying src/cryptography/hazmat/backends/multibackend.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends
                  creating build/lib.linux-x86_64-2.7/cryptography/hazmat/bindings/openssl
                  copying src/cryptography/hazmat/bindings/openssl/binding.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/bindings/openssl
                  copying src/cryptography/hazmat/bindings/openssl/_conditional.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/bindings/openssl
                  copying src/cryptography/hazmat/bindings/openssl/__init__.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/bindings/openssl
                  creating build/lib.linux-x86_64-2.7/cryptography/hazmat/bindings/commoncrypto
                  copying src/cryptography/hazmat/bindings/commoncrypto/binding.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/bindings/commoncrypto
                  copying src/cryptography/hazmat/bindings/commoncrypto/__init__.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/bindings/commoncrypto
                  creating build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/kdf
                  copying src/cryptography/hazmat/primitives/kdf/hkdf.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/kdf
                  copying src/cryptography/hazmat/primitives/kdf/pbkdf2.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/kdf
                  copying src/cryptography/hazmat/primitives/kdf/x963kdf.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/kdf
                  copying src/cryptography/hazmat/primitives/kdf/__init__.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/kdf
                  copying src/cryptography/hazmat/primitives/kdf/kbkdf.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/kdf
                  copying src/cryptography/hazmat/primitives/kdf/concatkdf.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/kdf
                  copying src/cryptography/hazmat/primitives/kdf/scrypt.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/kdf
                  creating build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/asymmetric
                  copying src/cryptography/hazmat/primitives/asymmetric/rsa.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/asymmetric
                  copying src/cryptography/hazmat/primitives/asymmetric/__init__.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/asymmetric
                  copying src/cryptography/hazmat/primitives/asymmetric/utils.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/asymmetric
                  copying src/cryptography/hazmat/primitives/asymmetric/dsa.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/asymmetric
                  copying src/cryptography/hazmat/primitives/asymmetric/dh.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/asymmetric
                  copying src/cryptography/hazmat/primitives/asymmetric/ec.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/asymmetric
                  copying src/cryptography/hazmat/primitives/asymmetric/padding.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/asymmetric
                  creating build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/twofactor
                  copying src/cryptography/hazmat/primitives/twofactor/__init__.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/twofactor
                  copying src/cryptography/hazmat/primitives/twofactor/utils.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/twofactor
                  copying src/cryptography/hazmat/primitives/twofactor/totp.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/twofactor
                  copying src/cryptography/hazmat/primitives/twofactor/hotp.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/twofactor
                  creating build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/ciphers
                  copying src/cryptography/hazmat/primitives/ciphers/algorithms.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/ciphers
                  copying src/cryptography/hazmat/primitives/ciphers/__init__.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/ciphers
                  copying src/cryptography/hazmat/primitives/ciphers/base.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/ciphers
                  copying src/cryptography/hazmat/primitives/ciphers/modes.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/ciphers
                  creating build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/interfaces
                  copying src/cryptography/hazmat/primitives/interfaces/__init__.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/primitives/interfaces
                  creating build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/openssl
                  copying src/cryptography/hazmat/backends/openssl/rsa.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/openssl
                  copying src/cryptography/hazmat/backends/openssl/hashes.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/openssl
                  copying src/cryptography/hazmat/backends/openssl/__init__.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/openssl
                  copying src/cryptography/hazmat/backends/openssl/decode_asn1.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/openssl
                  copying src/cryptography/hazmat/backends/openssl/utils.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/openssl
                  copying src/cryptography/hazmat/backends/openssl/encode_asn1.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/openssl
                  copying src/cryptography/hazmat/backends/openssl/dsa.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/openssl
                  copying src/cryptography/hazmat/backends/openssl/ciphers.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/openssl
                  copying src/cryptography/hazmat/backends/openssl/dh.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/openssl
                  copying src/cryptography/hazmat/backends/openssl/backend.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/openssl
                  copying src/cryptography/hazmat/backends/openssl/hmac.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/openssl
                  copying src/cryptography/hazmat/backends/openssl/ec.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/openssl
                  copying src/cryptography/hazmat/backends/openssl/x509.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/openssl
                  copying src/cryptography/hazmat/backends/openssl/cmac.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/openssl
                  creating build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/commoncrypto
                  copying src/cryptography/hazmat/backends/commoncrypto/hashes.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/commoncrypto
                  copying src/cryptography/hazmat/backends/commoncrypto/__init__.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/commoncrypto
                  copying src/cryptography/hazmat/backends/commoncrypto/ciphers.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/commoncrypto
                  copying src/cryptography/hazmat/backends/commoncrypto/backend.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/commoncrypto
                  copying src/cryptography/hazmat/backends/commoncrypto/hmac.py -> build/lib.linux-x86_64-2.7/cryptography/hazmat/backends/commoncrypto
                  running egg_info
                  writing requirements to src/cryptography.egg-info/requires.txt
                  writing src/cryptography.egg-info/PKG-INFO
                  writing top-level names to src/cryptography.egg-info/top_level.txt
                  writing dependency_links to src/cryptography.egg-info/dependency_links.txt
                  writing entry points to src/cryptography.egg-info/entry_points.txt
                  reading manifest file 'src/cryptography.egg-info/SOURCES.txt'
                  reading manifest template 'MANIFEST.in'
                  no previously-included directories found matching 'docs/_build'
                  warning: no previously-included files matching '*' found under directory 'vectors'
                  writing manifest file 'src/cryptography.egg-info/SOURCES.txt'
                  running build_ext
                  generating cffi module 'build/temp.linux-x86_64-2.7/_padding.c'
                  creating build/temp.linux-x86_64-2.7
                  generating cffi module 'build/temp.linux-x86_64-2.7/_constant_time.c'
                  generating cffi module 'build/temp.linux-x86_64-2.7/_openssl.c'
                  building '_openssl' extension
                  creating build/temp.linux-x86_64-2.7/build
                  creating build/temp.linux-x86_64-2.7/build/temp.linux-x86_64-2.7
                  gcc -pthread -fno-strict-aliasing -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/include/python2.7 -c build/temp.linux-x86_64-2.7/_openssl.c -o build/temp.linux-x86_64-2.7/build/temp.linux-x86_64-2.7/_openssl.o
                  build/temp.linux-x86_64-2.7/_openssl.c:434:30: fatal error: openssl/opensslv.h: No such file or directory
                   #include <openssl/opensslv.h>
                                                ^
                  compilation terminated.
                  error: command 'gcc' failed with exit status 1

                  ----------------------------------------   Failed building wheel for cryptography
                Failed cleaning build dir for cryptography
              Command "/usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-ux0MHT/cryptography/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-MmCJjn-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-ux0MHT/cryptography/
     Started: 15:25:55.194519
    Duration: 24527.827 ms
     Changes:
```

**Comment by rajvidhimar** [comment]
There are some additional dependencies which are needed to be added according to the OS. I'll add those. But from what I gather, the tests are run on multiple OSes. PyEZ (required to run junos tests) is supported on a limited number of OSes (Fedora, CentOS, Debian, Ubuntu, FreeBSD, MacOS). Is there a way to disable running junos unit tests on a  particular OS?

**Comment by twangboy** [comment]
@rajvidhimar You can use the `@SkipIf` decorator before each test you want to skip. Look at other tests for some examples.

**Comment by rallytime** [comment]
@rajvidhimar Yes, @twangboy is correct. You can skip individual tests or entire test classes based on certain conditions. (Like if `PyEZ` is available on the test box.) This would need to be added to the test class itself in your salt PR.

**Comment by rallytime** [comment]
@rajvidhimar Awesome! Thank you for updating this. This looks great from my perspective, but I want to make sure @twangboy reviews it to make sure we're good on the windows side with this change.

@twangboy Can you review this and let me know if this is OK to merge in from your perspective?

File: python/junos-eznc.sls [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{% if grains['os'] not in ('Windows') %}
```
Comment (twangboy) [inline python/junos-eznc.sls]: I think this is working on Windows now.

File: git/salt.sls [hunk 2]
```diff
@@ -331,4 +335,4 @@ install-salt-pytest-pip-deps:
   pip.installed:
     - requirements: /testing/requirements/pytest.txt
     - onlyif: '[ -f /testing/requirements/pytest.txt ]'
-{%- endif %}
+{%- endif %}
```
Comment (twangboy) [inline git/salt.sls]: Need new line here

File: python/junos-eznc.sls [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+{% endif %}
+
+junos-eznc:
+  pip.installed:
+    {%- if salt['config.get']('virtualenv_path', None)  %}
+    - bin_env: {{ salt['config.get']('virtualenv_path') }}
+    {%- endif %}
+    - index_url: https://pypi-jenkins.saltstack.com/jenkins/develop
+    - extra_index_url: https://pypi.python.org/simple
+{% if grains['os'] not in ('Windows') %}
```
Comment (twangboy) [inline python/junos-eznc.sls]: Also no need to gate here

File: python/jxmlease.sls [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{% if grains['os'] not in ('Windows') %}
```
Comment (twangboy) [inline python/jxmlease.sls]: This also works on Windows

File: python/jxmlease.sls [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+{% endif %}
+
+jxmlease:
+  pip.installed:
+    {%- if salt['config.get']('virtualenv_path', None)  %}
+    - bin_env: {{ salt['config.get']('virtualenv_path') }}
+    {%- endif %}
+    - index_url: https://pypi-jenkins.saltstack.com/jenkins/develop
+    - extra_index_url: https://pypi.python.org/simple
+{% if grains['os'] not in ('Windows') %}
```
Comment (twangboy) [inline python/jxmlease.sls]: This can also be removed.

**Comment by rallytime** [comment]
@twangboy Can you swing by one more time here on this review?

**Comment by rallytime** [comment]
Thanks for your work here @rajvidhimar!

**Pull request merged**